onnxruntime/core/framework/sequential_executor.cc

```diff
@@ -149,12 +149,18 @@ Status SequentialExecutor::Execute(const SessionState& session_state, const std:
     {
       diagnostic::span span(series, "%s.%d", node.OpType().c_str(), node.Index());
 #endif
+      Status compute_status;
+
+      try {
+        compute_status = p_op_kernel->Compute(&op_kernel_context);
```

Comment:
Actually I had proposed this try...catch logic a while back as part of improving error messages. But at that time we felt encapsulating the Compute() call within a try...catch might add overhead leading to perf regressions. @pranavsharma ?

Reply:
If high-verbosity, can we ignore perf issues and be as verbose as possible?

Reply:
Yeah, we already have a top level try-catch. I'm not sure if having a try-catch for every single node introduces any perf impact; of course we won't know without measuring it. It does seem a bit of an overkill given that most of the time there might not be any issues.

Reply:
Main benefit is you get the failing node name. Alternative is to re-work all the code that uses ORT_ENFORCE inside of compute to return a status instead. 

I believe most compilers these days use table based lookup for exception handling, so there's no perf cost on the non-throw path. x86 may differ and use the older style approach where there was a perf cost (would need to build release x86 and check the generated assembly).

Reply:
May be we can do this conditionally - based on debug flag or log verbosity level? Also, fyi - in a recent binary size analysis exception data made up for 300-400K (don't remember exact size). Visual studio 19 has [much better support](https://devblogs.microsoft.com/cppblog/making-cpp-exception-handling-smaller-x64/) for exceptions.

Reply:
Assembly generated by VC++ for x64 is exactly the same for the call with and without the try/catch so no extra instructions on the no exception path. x86 has two extra instructions to jump over the catch. I wouldn't expect that to have any impact of performance, so I'm wondering does that really warrant only conditionally having the try/catch?

Reply:
Yeah, in this case, it's fine.